### PR TITLE
:memo: AlpineJS Type Documentation and Touchups

### PR DESCRIPTION
**Pull request by ekwoka**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: This is improving the accuracy of types to cover more edge cases
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

Supports strict function checks when data methods have args, looses up some types that were overly strict. Adds some JSDoc info to some methods.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/alpinejs/index.d.ts
1. types/alpinejs/alpinejs-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against a393dc6680577d6fe524b875d5b6698695e94d08
</p>

**Review by wtho — APPROVED** [review]
Thanks for improving the types and especially putting so much effort with the documentation!

Sorry for letting you wait with the review.

**Comment by ekwoka** [comment]
Ready to merge

File: types/alpinejs/index.d.ts [hunk 1]
```diff
@@ -103,10 +103,10 @@ export interface InterceptorObject<T> {
...
 
 type InferInterceptor<T> = T extends InterceptorObject<infer U> ? U
-    : T extends Record<string | number | symbol, unknown> ? {
-            [K in keyof T]: InferInterceptor<T[K]>;
-        }
-    : T;
+    : keyof T extends never ? T
+    : {
+        [K in keyof T]: InferInterceptor<T[K]>;
+    };
```
Comment (mickaelchanrion) [inline types/alpinejs/index.d.ts:109]: @ekwoka I noticed that this change doesn't let me store complex object in component.
If install the version @types/alpinejs@3.13.5, this problem goes away:
image

Comment (ekwoka) [reply types/alpinejs/index.d.ts:109]: 😭 

If it wasn't for interceptors being how they are (I didn't make them don't blame me), this would be so easy...

Maybe, for now, this can just drop support for nested interceptors....

but then I use nested interceptors in projects....maybe limit the depth a certain amount?...

To at least reduce the potential conflicts...

I'll try to get this figured out.

Do you have like the full TS error? Like where is the spot that it determines as being "not assignable"? I assume it's because there is some kind of `any` somewhere....

Comment (ekwoka) [reply types/alpinejs/index.d.ts:109]: #69050 should solve this (or at least did In my checks...)

Comment (mickaelchanrion) [reply types/alpinejs/index.d.ts:109]: I just tested on my side, it's solved!
Thank you so much for this quick fix, you rock! 💯
